### PR TITLE
[7238] Save and minimize then save and close in a new embeded content cause an error

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/7238

### CODE REVIEW

File: static-assets/components/cstudio-forms/data-sources/embedded-content.js [hunk 1]
```diff
@@ -39,6 +40,7 @@ YAHOO.extend(CStudioForms.Datasources.EmbeddedContent, CStudioForms.CStudioFormD
   itemsAreContentReferences: true,
 
   createElementAction: function (control, _self) {
+    _self.inserted = false;
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/embedded-content.js:43]: Should this be cleared at some point or is the instantiation lifecycle enough to manage that state?

File: static-assets/components/cstudio-forms/data-sources/embedded-content.js [hunk 2]
```diff
@@ -66,7 +68,16 @@ YAHOO.extend(CStudioForms.Datasources.EmbeddedContent, CStudioForms.CStudioFormD
...
+            if (!_self.inserted) {
+              control.insertItem(name, value, null, null, _self.id);
+              _self.inserted = true;
+            } else {
+              // Recently added item is in the last position
+              const itemIndex = control.items.length - 1;
+              // When adding a new embedded content, the form may be saved using the 'Save draft' or the 'Save & Minimize'
+              // options. When it happens, the next save operation will be an edition, not a creation.
+              control.updateEditedItem({ value }, _self.id, itemIndex);
+            }
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/embedded-content.js:89]: - What about the path above (the if case)?
- I'm not completely in context, but not sure the last position will always contain the item that's being created or edited?

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-forms/data-sources/embedded-content.js (hunk 2 above)
Comment (jvega190) [reply static-assets/components/cstudio-forms/data-sources/embedded-content.js:89]: In this case (when creating a new node-selector content) it'll always be at the last position. Also, the update call in here will refer only to the newly created item, that again, will be at the last position

**Comment by jvega190** [comment]
Closing this PR as we opted for a different approach

**Pull request closed without merging**